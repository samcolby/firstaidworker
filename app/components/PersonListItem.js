import React, { PureComponent } from "react";
import PropTypes from "prop-types";

import { ListItem } from "react-native-elements";
import { FastImage } from "react-native-fast-image";

/**
 * React Pure component used to display a person's
 * information on the row of a flat list.
 *
 * This is used on the People Near Me tab.
 *
 * @class PersonListItem
 * @extends {PureComponent}
 */
class PersonListItem extends PureComponent {
  static propTypes = {
    person: PropTypes.object.isRequired,
    onPress: PropTypes.func
  };

  constructor(props) {
    super(props);
    this.onPress = this.onPress.bind(this);
  }

  onPress() {
    this.props.onPress(this.props.person);
  }

  render() {
    const { name, job_title, avatar_uri } = this.props.person;

    return (
      <ListItem
        leftAvatar={{
          rounded: true,
          source: { uri: avatar_uri },
          title: name[0],
          ImageComponent: FastImage
        }}
        onPress={this.onPress}
        subtitle={job_title}
        title={name}
      />
    );
  }
}

export default PersonListItem;
