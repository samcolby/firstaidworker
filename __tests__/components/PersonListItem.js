// __tests__/Intro-test.js
import React from "react";

import { shallow } from "enzyme";

import PersonListItem from "../../app/components/PersonListItem";

const testData = {
  id: "c4ab9532-198e-4891-8719-8167005e9fa9",
  name: "Maryellen Baxter",
  job_title: "Flouncer",
  isActive: true,
  picture: "https://randomuser.me/api/portraits/women/12.jpg",
  company: "PANZENT",
  email: "maryellenbaxter@panzent.com",
  phone: "+1 (848) 545-3302",
  latitude: 51.2000548,
  longitude: -4.1474755
};

describe("<PersonListItem /> component", () => {
  it("renders correctly", () => {
    const component = shallow(<PersonListItem person={testData} />);

    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it("displays the person's name correctly", () => {
    const component = shallow(<PersonListItem person={testData} />);
    expect(component.prop("title")).toEqual(testData.name);
  });

  it("displays the person's picture", () => {
    const component = shallow(<PersonListItem person={testData} />);
    expect(component.prop("leftAvatar").source.uri).toEqual(testData.picture);
  });

  it("handles onPress", () => {
    const onPress = jest.fn();
    const component = shallow(
      <PersonListItem person={testData} onPress={onPress} />
    );

    component.simulate("press");
    expect(onPress).toHaveBeenCalled();
    expect(component).toMatchSnapshot();
  });
});
