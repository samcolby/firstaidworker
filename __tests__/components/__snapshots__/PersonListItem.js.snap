// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonListItem /> component handles onPress 1`] = `
<Themed.ListItem
  leftAvatar={
    Object {
      "rounded": true,
      "source": Object {
        "uri": "https://randomuser.me/api/portraits/women/12.jpg",
      },
    }
  }
  onPress={[Function]}
  subtitle="Flouncer"
  title="Maryellen Baxter"
/>
`;

exports[`<PersonListItem /> component renders correctly 1`] = `
<Themed.ListItem
  leftAvatar={
    Object {
      "rounded": true,
      "source": Object {
        "uri": "https://randomuser.me/api/portraits/women/12.jpg",
      },
    }
  }
  onPress={[Function]}
  subtitle="Flouncer"
  title="Maryellen Baxter"
/>
`;
